docs/madr/1-intial.md

```diff
@@ -0,0 +1,155 @@
+# An event driven ANT registration system in AO to optimize ANT listings by access role
+
+- Status: accepted
+- Deciders: [Dylan], [Ariel], [Phil]
+- Date: 2024-07-22
+- Authors: [Atticus], [Dylan], [Ariel], [Phil]
+
+## Context and Problem Statement
+
+Listing all ANTs registered to ArNS names and owned by a specific owner or
+owners becomes more computationally expensive as more ArNS names are registered.
+Additionally, its computationally infeasible to find ANTs not registered to an
+ArNS. GraphQL-based approaches cannot currently support either of these
+scenarios.
+
+We need to provide owner-based lookups that perform better than O(N).
+
+This is done both on the client side via [AoConnect], or as inter-process
+communication (like in the [ArNS Resolver](#arns-resolver)).
+
+### Additional Background
+
+The ArNS AO Process is responsible for recording which ANTs are registered to
+each ArNS name. It does not store or track additional information about the
+owner of the ANT. This reduces the amount of state managed, providing more
+concurrency in the operations in the overall name system's design, (e.g.
+individual ANT's activities do not impact performance of the ArNS Process).
+
+The downside of this separation is that listing all the ANTs for any owner is
+computationally expensive. You need to retrieve every ArNS record and perform a
+lookup of each associated ANT's owner, then filter the resulting owner map by
+the owner(s) in question - an O(N) operation where N is the total number of
+registered names. An additional downside is that unregistered (from an ArNS
+perspective) ANTs that one owns are not discoverable via this expensive
+algorithm.
+
+## Decision Drivers
+
+The main drivers for this decision revolve around:
+
+- **Speed**
+  - Uses fewer resources client-side.
+- **Scalability**
+  - We cannot load the entire ArNS ecosystem client-side if it grows.
+- **Ecosystem Integrations**
+  - Having this utility process solely for ANT contract indexing provides
+    integration points for other applications and reduces our reliance on other
+    teams/products (e.g., Bazar profile processes).
+- **Maintainability**
+  - Since we control the registry process, we can drive its maintenance.
+
+## Considered Options
+
+### Dedicated ANT Registry Process
+
+An AO Process that allows for permissionless reporting of the existence of ANTs
+by their Process ID that is capable of asynchronously, by way of AO messaging,
+retrieving, or being notified about, and then caching indexable information
+about ANTs, namely their owners and Controllers.
+
+- **Pros:**
+  - Dedicated registry for ANTs.
+  - Event-driven updates from ANTs.
+  - Separate state and compute from the resolver process, thus preserving system
+    concurrency while providing for better UX in the system.
```

Comment:
this is suggesting the resolver is a permanent part of the stack. I'd say opportunity to integrate full name resolution via a separate process or internally.

Reply:
I don't understand what you mean.